content/install-user-css.js

```diff
@@ -0,0 +1,284 @@
+            $element({tag: 'small', className: 'meta-version', textContent: style.version})
+          ]}),
+          $element({tag: 'p', textContent: style.description}),
+          $element({tag: 'h3', textContent: t('author')}),
+          style.author,
+          $element({tag: 'h3', textContent: t('license')}),
+          style.license,
+          $element({tag: 'h3', textContent: t('appliesLabel')}),
+          $element({tag: 'ul', appendChild: getAppliesTo(style).map(
+            pattern => $element({tag: 'li', textContent: pattern})
+          )}),
+          $element({className: 'actions', appendChild: [
+            $element({tag: 'button', className: 'install', textContent: installButtonLabel()})
+          ]}),
+          $element({className: 'external', appendChild: [
+            externalLink('externalHomepage', style.url),
+            externalLink('externalSupport', style.support)
+          ]})
+        ]}),
+        $element({className: 'main', appendChild: [
+          $element({className: 'code'})
+        ]})
+      ]});
+    }
+
+    function externalLink(name, url) {
+      return $element({
+        tag: 'a',
+        href: url,
+        target: '_blank',
+        textContent: t(name),
+        rel: 'noopener'
+      });
+    }
+
+    function installButtonLabel() {
+      return t(!dup ? 'installButton' :
+        versionTest > 0 ? 'installButtonUpdate' : 'installButtonReinstall');
+    }
+  });
+}
+
+function initLiveReload(sourceLoader) {
+  let installed;
+  const watcher = sourceLoader.watch(source => {
+    $('.code').textContent = source;
+    return runtimeSend({
+      method: 'saveUsercss',
+      id: installed.id,
+      source: source
+    }).then(() => {
+      $$('.main .warning').forEach(e => e.remove());
+    }).catch(err => {
+      const oldWarning = $('.main .warning');
+      const warning = buildWarning(err);
+      if (oldWarning) {
+        oldWarning.replaceWith(warning);
+      } else {
+        $('.main').insertBefore(warning, $('.main').childNodes[0]);
+      }
+    });
+  });
+  window.addEventListener('installed', ({detail: {style}}) => {
+    installed = style;
+    if ($('.live-reload-checkbox').checked) {
+      watcher.start();
+    }
+  });
+  $('.actions').appendChild($element({tag: 'label', className: 'live-reload', appendChild: [
+    $element({tag: 'input', type: 'checkbox', className: 'live-reload-checkbox'}),
+    $element({tag: 'span', textContent: t('liveReloadLabel')})
+  ]}));
+  $('.live-reload-checkbox').onchange = e => {
+    if (!installed) {
+      return;
+    }
+    if (e.target.checked) {
+      watcher.start();
+    } else {
+      watcher.stop();
+    }
+  };
+}
+
+function buildWarning(err) {
+  return $element({className: 'warning', appendChild: [
+    t('parseUsercssError'),
+    $element({tag: 'pre', textContent: String(err)})
+  ]});
+}
+
+function initErrorPage(err, source) {
+  return pendingResource.then(() => {
+    document.body.innerHTML = '';
+    [
```

Comment:
We need to avoid using `innerHTML` because of AMO.

Reply:
Yep, `textContent = '';`